beps/bep_attrs.rst

```diff
@@ -0,0 +1,119 @@
+:BEP: ??
+:Title: Padding files and extended file attributes 
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  The 8472 <the8472.bep@infinite-source.de>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 05-Aug-2016
+:Post-History: 
+
+
+Padding files and extended file attributes
+==========================================
+
+This BEP specifies some additional file properties beyond those described those in BEP 3 [#BEP-3]_.
+
+
+Multi-file format
+
+.. parsed-literal::
+
+    {
+      "info":
+      {
+        "files":
+        {[
+          {
+            "attr": "phxl",
+            "sha1": *<20 bytes>*,
+            "symlink path": ["dir1", "dir2", "target.ext"],
+            ...
+          },
+          {
+            ...
+          }
+        ]},
+        ...
+      },
+      ...
+    }
+    
+Single-file format
+
+.. parsed-literal::
+
+
+    {
+      "info":
+      {
+        "attr": "hx",
+        "sha1": *<20 bytes>*,
+        ...
+      },
+      ...
+    }
+
+
+
+
+``attr``
+  A variable-length string. When present the characters each represent a file attribute. l = symlink, x = executable, h = hidden, p = padding file. Characters appear in no particular order and unknown characters should be ignored.
+  
+``sha1``
+  20 bytes. The SHA1 digest calculated over the contents of the file itself, without any additional padding. Can be used to aid file deduplication [#BEP-38]_.
+  
+``symlink path``
+  An array of strings. Path of the symlink target relative to the torrent root directory.
+
+
+Symlinks
+========
+
+When the ``l`` attribute flag is present then the ``symlink path`` represents the *target* of the symlink while ``path`` indicates the location of the symlink itself.
+  
+The ``length`` field of the symlink file should be zero. A non-zero length identical with the target file would improve backwards-compatibility but significantly complicate the management of piece hashing and duplicate pieces.
+  
+The target should be another file within the torrent, otherwise a dangling symlink will be created.
```

Comment:
this is an interesting way to restrict this and avoid opening up dangerous holes from a download directory into any other location on the machine. I'm not sure libtorrent restricts this right now. Maybe it makes sense to explicitly spell out that the link target must be a relative path too then